pkg/kubernetes/scripts/virtual-machines/entry-points/virtual-machine.jsx

```diff
@@ -0,0 +1,80 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2018 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import 'regenerator-runtime/runtime'; // required for library initialization
+import React from 'react';
+import { Provider } from 'react-redux';
+
+import { setVms, vmExpanded } from '../action-creators.jsx'
+import VmDetail from '../components/VmDetail.jsx';
+import { initStore, getStore } from '../store.es6';
+import initialize from './util/initialize.es6';
+
+import '../../../../machines/machines.less'; // once per component hierarchy
+
+const VmPage = ({vmUid}) => (
+    <Provider store={getStore()}>
+        <VmDetail vmUid={vmUid} />
+    </Provider>
+);
+
+function addVmListener (store, $scope, kubeLoader, kubeSelect, vmUid) {
+    kubeLoader.listen(() => {
+        const vms = kubeSelect().kind('VirtualMachine')
+                .uid(vmUid)
+
+        store.dispatch(setVms(Object.values(vms)))
+    }, $scope)
+    kubeLoader.watch('VirtualMachine', $scope)
+}
+
+/**
+ *
+ * @param {$rootScope.Scope} $scope 'VirtualMachinesCtrl' controller scope
+ * @param {$routeParams} $routeParams
+ * @param {kubeLoader} kubeLoader
+ * @param {kubeSelect} kubeSelect
+ * @param {kubeMethods} kubeMethods
+ * @param {KubeRequest} KubeRequest
+ */
+function init($scope, $routeParams, kubeLoader, kubeSelect, kubeMethods, KubeRequest) {
```

Comment:
I think we can share can between these two entry points even more

Reply:
What do you suggest? Unifying the id of the root element and passing the component to render? I do not see other improvements otherwise.